src/main/java/com/codewitches/codefellowship/controllers/ApplicationUserController.java

```diff
@@ -46,4 +46,14 @@ public String showOneUser(@PathVariable long id, Principal p, Model m) {
         return "userProfile";
     }
 
-}
\ No newline at end of file
+    @GetMapping("/userProfile")
+    public String showUserProfile(Principal p, Model m) {
+        m.addAttribute("viewedUser", applicationUserRepository.findByUsername(p.getName()));
+        m.addAttribute("user", p);

```

Comment:
Food for thought: I you're passing the `ApplicationUser` object that represent the logged in user, do you need to pass the `Principal` as well?